code/modules/clothing/head/tinfoilhat.dm

```diff
@@ -67,3 +71,31 @@
 	. = ..()
 	if(!warped)
 		warp_up()
+
+//Man I love signals! They're my favorite!
+/obj/item/clothing/head/foilhat/proc/call_suicide(datum/source)
+	SIGNAL_HANDLER
+	suicide_act(source)
+
+/obj/item/clothing/head/foilhat/suicide_act(mob/living/user)
+	user.visible_message(span_suicide("[user] gets a crazed look in [user.p_their()] eyes! [capitalize(user.p_they())] [user.p_have()] witnessed the truth, and try to commit suicide!"))
+	var/suicide_type = INVOKE_ASYNC(src, /obj/item/clothing/head/foilhat/proc/suicide_user, user)	//SIGNAL_HANDLER doesn't like to wait, so i decided to move it to a proc in INVOKE_ASYNC
+	return suicide_type
```

Comment:
You can't expect a return from `INVOKE_ASYNC` - it's running separately from the proc. You'll need to just return whatever in this proc itself.